source/globalCommands.py

```diff
@@ -1786,7 +1785,7 @@ def script_review_copy(self, gesture):
 				pass
 			try:
 				copyMarker.updateSelection()
-				if isinstance(pos.obj, editableText.EditableTextWithoutAutoSelectDetection):
+				if hasattr(pos.obj, "waitForAndReportSelectionChange"):

```

Comment:
There are three possibilities for obj:
1. It's an EditableTextWithAutoSelectDetection. In that case, the selection change will be detected automatically and there's nothing to do here.
2. It's an EditableTextWithoutAutoSelectDetection, in which case we need to wait for and report the selection change here because it can't detect the change itself.
3. It's a CursorManager, in which case we need to report the change because it can't detect itself. However, we don't need to wait here because we trust it to update immediately.

The problem with moving the new code into the else clause is that it breaks case 1; it'll end up reporting the change twice. Or am I misunderstanding your suggestion?

I agree the method name seems wrong now. What about renaming it to just reportSelectionChange? Whether or not we need to wait to successfully report a change is then just an implementation detail.